### PR TITLE
refactor: make pkce option a client setting

### PR DESCRIPTION
**Pull request by J0**
Refactor pkce to be a client setting so it doesn't have to be configured on invocation for each method.

- also call `exchangeSessionForCode` in the background if `detectSession` is enabled
- also remove the `flowType` option from signInWithOAuth

### CODE REVIEW

File: src/GoTrueClient.ts [hunk 1]
```diff
@@ -920,6 +935,16 @@ export default class GoTrueClient {
         Boolean(getParameterByName('error_description')))
     )
   }
+  /**
+   * Checks if the current URL and backing storage contain parameters given by a PKCE flow
+   */
+  private async _isPKCEFlow(): Promise<boolean> {
+    const currentStorageContent = await getItemAsync(
+      this.storage,
+      `${this.storageKey}-code-verifier`
```
Comment (thorwebdev) [inline src/GoTrueClient.ts:944]:
```suggestion
      `${this.storageKey}-oauth-code-verifier`
```

**Review by thorwebdev — APPROVED** [review]
One storage naming issue in `_isPKCEFlow` otherwise LGTM, thanks.

File: src/GoTrueClient.ts (hunk 1 above)
Comment (J0) [reply src/GoTrueClient.ts:944]: Ah, good catch! Have renamed everything else to `code-verifier` instead so as to keep it generic

**Comment by J0** [comment]
Leaving this with a refactor: prefix so semver doesn't trigger another release as I'd like to get another three PRs in 
1. remove OAuth `flowType` param 
2. add magic link binding
3. add recover bindings
Before making another release

**Pull request merged**